mpl_finance.py

```diff
@@ -605,11 +605,20 @@ def candlestick2_ohlc(ax, opens, highs, lows, closes, width=4,
 
     colorup = mcolors.to_rgba(colorup, alpha)
     colordown = mcolors.to_rgba(colordown, alpha)
-    colord = {True: colorup, False: colordown}
-    colors = [colord[open < close]
-              for open, close in zip(opens, closes)
-              if open != -1 and close != -1]
-
+    colordoji = mcolors.to_rgba(colordoji, alpha)
+    # colord = {True: colorup, False: colordown}
+    # colors = [colord[open < close]
+              # for open, close in zip(opens, closes)
+              # if open != -1 and close != -1]
+    colors = []
+    for open, close in zip(opens, closes):
+        if open != -1 and close != -1:
+            if open < close:
+                colors.append(colorup)
+            elif open > close:
+                colors.append(colordown)
+            else:
```

Comment:
Do you want to use some kind of a tolerance here?   As written, it seems to me, that you will get your doji color only if the open and close are *exactly* equal (which in real life would be a very rare event indeed).  Is that what you want?  Did you run this code against any real data and are you happy with the way it works?

Reply:
Respected Sir,
Here are the answer to what you have asked me in the above conversation.
**Do you want to use some kind of a tolerance here?**
No. I do not want any tolerance as the doji candle I expected is to be gray in color. The growing or up candles in Green and the degrading or down candles in Red color. These look somewhat standard.
I, often, get doji candles. And that will help me in separating them from the up and down candles.
**Is that what you want? Did you run this code against any real data and are you happy with the way it works?**
Yes, I want that. I have tried it too. On my system I have made the changes with the library. And have got excellent experience viewing the candles through Matplotlib library mpl-finance.
Please have a look at this image:
![Fig_164](https://user-images.githubusercontent.com/33365560/69602321-24858100-103d-11ea-9eb3-5a56ca4afd9f.png)
I hope that your doubts are cleared.